realsense_camera/src/base_nodelet.cpp

```diff
@@ -1206,6 +1208,40 @@ namespace realsense_camera
     }
   }
 
+  /*
+   * Display warning for non-validated camera firmware
+  */
+  void BaseNodelet::checkFirmwareValidation(std::string validatedFirmware,
+        rs_device* rs_detected_device, std::string typeOfFirmware)
+  {
+    std::string currentFirmware = "";
+    if (typeOfFirmware == "CAMERA")
+    {
+      currentFirmware = rs_get_device_firmware_version(rs_detected_device, &rs_error_);
+      checkError();
+    }
+    if (typeOfFirmware == "ADAPTER")
+    {
+      currentFirmware = rs_get_device_info(rs_detected_device,
+            RS_CAMERA_INFO_ADAPTER_BOARD_FIRMWARE_VERSION, &rs_error_);
+      checkError();
+    }
+    if (typeOfFirmware == "MOTION_MODULE")
+    {
+      currentFirmware = rs_get_device_info(rs_detected_device,
+            RS_CAMERA_INFO_MOTION_MODULE_FIRMWARE_VERSION, &rs_error_);
+      checkError();
+    }
+
+    if (currentFirmware != validatedFirmware)
+    {
+      ROS_WARN_STREAM(typeOfFirmware << " firmware is " << currentFirmware <<
```

Comment:
Please use the nodelet_name_ member variable in the warning message so that the user knows which nodelet is printing the warning. This is a standard we follow in our nodelet. See other ROS_INFO_STREAM() statements for examples.